Fix NULL pointer error if rules label is missing

makusuko opened the pull request:
If no label is provided after the RULE keyword, EPANET crashes. This fixes the crash by returning an error in the case the label is missing.

michaeltryby commented:
@makusuko I think your branch is behind dev.  You need to fetch the changes and pull them into your branch.

makusuko commented:
@michaeltryby Ah... I created the branch from master and not from dev - duh! Now it should be in order. I also added NULL guard for freerules function to avoid double free should the function be called multiple times, and changed a couple of strcpy, sprintf and strcat calls to their *n* variants to ensure the buffer boundaries are not exceeded.
